braindecode/augmentation/transforms.py

```diff
@@ -1070,3 +1072,61 @@ def get_augmentation_params(self, *batch):
             "lam": lam,
             "idx_perm": idx_perm,
         }
+
+
+class SegmentationReconstruction(Transform):
+    """Applies a segmentation-reconstruction transform to the input data, as
+       proposed in _[1].
```

Comment:
I guess we could say slightly more about what the transform does: creates synthetic EEG by randomly mixing segments of recordings of the same class from the batch.